partner/partner-management-service/src/main/java/io/mosip/pms/partner/misp/service/impl/InfraProviderServiceImpl.java

```diff
@@ -259,51 +260,65 @@ public MISPLicenseResponseDto regenerateKey(String mispId) {
 			throw new MISPServiceException(MISPErrorMessages.MISP_IS_INACTIVE.getErrorCode(),
 					MISPErrorMessages.MISP_IS_INACTIVE.getErrorMessage());
 		}
-		List<MISPLicenseEntity> mispLicenses = mispLicenseRepository.findByMispId(mispId);
-		boolean isActiveLicenseExists = false;
+		List<MISPLicenseEntity> mispValidLicenses = mispLicenseRepository.findByMispIdandExpirydate(mispId);
+		if(mispLicenseRepository.findByMispId(mispId).isEmpty()) {
+			throw new MISPServiceException(MISPErrorMessages.MISP_LICENSE_KEY_NOT_ASSOCIATED_MISP_ID.getErrorCode(),
+					MISPErrorMessages.MISP_LICENSE_KEY_NOT_ASSOCIATED_MISP_ID.getErrorMessage());
+		}
+		List<PartnerPolicyRequest> approvedPolicyMappedReq = partnerPolicyRequestRepository.findByPartnerId(mispId);
```

Comment:
Does this query return ordered entries? When an existing MISP partner is assigned with a new policy how does this behave? Will this query always return the latest mapped policy?

Reply:
as of now, we are expecting only 1 policy mapping has to be present for misp partner. For the issues in this regard, we already got a story created and gave it to BA to comeup with the approach on how to fix it.